src/app/offer/offer.component.ts

```diff
@@ -50,6 +52,18 @@ export class OfferComponent {
         ),
       )),
     )
+    this.links$ = this.campaign$.pipe(
+      filter((campaign) => !!campaign.value),
+      switchMap((campaign) => {
+        const previewLinks = campaign.value!.newsURLs?.map((url) => this.linkPreviewService.previewLink(url))
+
+        if (previewLinks && previewLinks.length > 0) {
+          return withStatus(combineLatest(previewLinks))
+        } else {
+          return of(undefined)
```

Comment:
why not an empty array?

Reply:
This way I can use `<ng-container *ngIf="links$ | async as linksRes">` as a conditional and not display "Project News" block at all if there are no links for the offer.

Reply:
yes, this is a usual problem with checking with ngif + async. in this case, you can wrap a result in an object like `{value: []}` with map operator (`map(value => ({value}))`)
this way, you can check if you actually received a response (the whole object will be `truthy`). if the value is an array, you can check if it's empty and show an appropriate message. I've done it before in some cases where I expected an empty response, you can check inside the project.

Reply:
I played around with it a bit but I didn't get it to behave correctly. The case here is this: if we have some links to return, we want to show the loading spinner. So we use `Observable<WithStatus<LinkPreviewResponse[]>>`. However, if we have no links to display, we return `Observable<undefined>` because there is nothing to load at all. Returning an empty array would mean we use `Observable<WithStatus<[]>>` but for this case we will never have loading. With that map call we could return `Observable<{ value: WithStatus<LinkPreviewResponse[]> | undefined }>`. But returning an empty array would not help here since then we would return `Observable<{ value: WithStatus<LinkPreviewResponse[]> | [] }>`. This also means we now have to use one more `value` accessor in HTML. I may be missing something here, but `Observable<WithStatus<LinkPreviewResponse[]> | undefined>` does the job of conditional displaying here quite well.